### PR TITLE
Gorgoris/dao/invoice interface test

### PR DESCRIPTION
**Pull request by metagon**

### CODE REVIEW

File: src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestInvoiceDAO.java [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+        invoiceDAO.create(invoice);
+
+        // THEN
+        // check if identity was set
+        assertNotNull(invoice.getIdentity());
+
+        Invoice matcher = new Invoice();
+        matcher.setIdentity(invoice.getIdentity());
+
+        List<Invoice> invoiceList = invoiceDAO.find(matcher);
```
Comment (emmanuel099) [inline src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestInvoiceDAO.java]: remove matcher and use Invoice.withIdentity instead

dao.find(Invoice.widthIdentity(...));

File: src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestInvoiceDAO.java [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+            invoiceDAO.create(invoice);
+        } catch (DAOException e) {
+            fail("DAOException was thrown instead");
+        } finally {
+            assertNull(invoice.getIdentity());
+        }
+    }
+
+    @Test(expected = ValidationException.class)
+    public void testCreate_shouldFailWithNegativeGrossAmount() throws ValidationException {
```
Comment (emmanuel099) [inline src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestInvoiceDAO.java]: please add a validator test with such test cases (no gross, negative gross, ....) and remove them from this test

File: src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestInvoiceDAO.java [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+    @Test
+    public void testUpdate_shouldUpdateObject() throws ValidationException, DAOException {
+        // GIVEN
+        Invoice invoice = new Invoice();
+        invoice.setTime(new Date());
+        invoice.setGross(26.7d);
+        invoiceDAO.create(invoice);
+
+        // Make sure the invoice was stored successfully
+        assertNotNull(invoice.getIdentity()); // Check if identity was set
```
Comment (emmanuel099) [inline src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestInvoiceDAO.java]: should only be tested in testCreate_shouldAddObject

File: src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestInvoiceDAO.java [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+        Invoice matcher = new Invoice();
+        matcher.setIdentity(invoice.getIdentity());
+        assertEquals(invoice, invoiceDAO.find(matcher).get(0));
```
Comment (emmanuel099) [inline src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestInvoiceDAO.java]: matcher not needed, use find(invoice)
and use assertEquals(1, dao.find(invoice).size()) instead (get will throw an IndexOutOfBoundsException when the list is empty)

File: src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestInvoiceDAO.java [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+
+    @Test(expected = ValidationException.class)
+    public void testUpdate_shouldFailWhenObjectIsNull() throws ValidationException {
+        // GIVEN
+        Invoice invoice = null;
+
+        // WHEN
+        try {
+            invoiceDAO.update(invoice);
+        } catch (DAOException e) {
```
Comment (emmanuel099) [inline src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestInvoiceDAO.java]: not needed, junit will complain when an unexpected exception has been thrown - applies for all test cases

File: src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestInvoiceDAO.java [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+        // WHEN
+        try {
+            invoiceDAO.delete(invoice);
+        } catch (DAOException e) {
+            fail("DAOException was thrown instead");
+        }
+    }
+
+    @Test
+    public void testFind_shouldFindObject() throws ValidationException, DAOException {
```
Comment (emmanuel099) [inline src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestInvoiceDAO.java]: add an additional test case testFind_byWaiterShouldReturnObjects
